client/lib/resize-image-url/index.js

```diff
@@ -61,32 +61,35 @@ const scaleByFactor = value => value * IMAGE_SCALE_FACTOR;
  * URL containing query string arguments is passed to this function, it will
  * return `null`.
  *
- * @param   {String}          imageUrl Original image url
- * @param   {(Number|Object)} resize   Resize pixel width, or object of query
+ * @param   {string}          imageUrl Original image url
+ * @param   {(number|object)} resize   Resize pixel width, or object of query
  *                                     arguments (assuming Photon or Gravatar)
- * @param   {?Number}         height   Pixel height if specifying resize width
- * @param   {?Boolean}        makeSafe Should we make sure this is on a safe host?
- * @returns {?String}                  Resized image URL, or `null` if unable to resize
+ * @param   {?number}         height   Pixel height if specifying resize width
+ * @param   {?boolean}        makeSafe Should we make sure this is on a safe host?
+ * @returns {?string}                  Resized image URL, or `null` if unable to resize
  */
 export default function resizeImageUrl( imageUrl, resize, height, makeSafe = true ) {
 	if ( 'string' !== typeof imageUrl ) {
 		return imageUrl;
 	}
 
-	const parsedUrl = parse( imageUrl, true, true );
-	if ( ! REGEXP_VALID_PROTOCOL.test( parsedUrl.protocol ) ) {
+	const urlType = determineUrlType( imageUrl );
+
+	if ( urlType !== URL_TYPE.ABSOLUTE ) {
+		// Invalid URL; return unmodified.
 		return imageUrl;
 	}
-	if ( ! parsedUrl.hostname ) {
-		// no hostname? must be a bad url.
+
+	const parsedUrl = new URL( imageUrl, 'https://__domain__.invalid' );
+
+	if ( ! REGEXP_VALID_PROTOCOL.test( parsedUrl.protocol ) ) {
```

Comment:
exception risk, and unnecessary RegExp after the change…

```js
const url = parseURL( imageUrl, 'https://__domain__.invalid' );

if ( ! ( url && ( 'https:' === url.protocol || 'http:' === url.protocol ) ) ) {
	return imageUrl;
}
```

Reply:
There's no exception risk here, because we already tested above that we don't have an invalid URL (we'd get `URL_TYPE.INVALID` on the `urlType` if that were the case).